### DISCUSSION TITLE
DAG infinite loop

### DISCUSSION THREAD

**Discussion by vivisabbi** [apache/airflow#28989] [Q&A]
Hi

My DAG is keep executing continuously with out respecting the schedule given. What could be the issue.

Thanks
Vivek

**Comment by josh-fell** [comment]
Hi @vivisabbi! Can you provide more context and/or a code snippet of your DAG definition? More details will make helping you much, much easier. 

My first thought/question is: is `catchup=False` in your DAG? By default, Airflow will attempt to fill in DAG runs from your defined `start_date` to the current interval. If the `start_date` is in the past, the DAG runs will continue to be created from that data interval until the present time.

**Reply by vivisabbi** [reply]
Hi Josh, Thanks for the providing information which is very useful to know. Below is my DAG definition code.  

default_args = {
        'owner': 'airflow',
        'retries': 5,
        'retry_delay': timedelta(minutes=2)
}

with DAG(
        dag_id='devsecops',
        default_args=default_args,
        description='this is simulation dag of airflow integration with devsecops',
        start_date=datetime(2022,12,14,5),
        default_view='graph',
        schedule_interval='@daily'
        )as dag:

**Reply by vivisabbi** [reply]
Also my DAG  running twice for some reason. what could be the reason, 

Thanks in advance

**Reply by josh-fell** [reply]
Can you try setting `catchup=False` in your DAG definition? Otherwise, DAG runs will be created for each daily interval from 2022-12-14 until the current interval. I assume this what you mean by "running continuously".

[Here is some documentation on the catchup functionality](https://docs.astronomer.io/learn/rerunning-dags#catchup) should you find it useful.

**Reply by vivisabbi** [reply]
Thank you Josh, I  will update my DAG definition.
